internal/core/cppapi_data.cpp

```diff
@@ -417,4 +417,70 @@ void attributes(
     calc_attributes(horizon, surface, src_window, dst_window, attrs, from, to);
 }
 
+void align_surfaces(
+    RegularSurface const &primary,
+    RegularSurface const &secondary,
+    RegularSurface &aligned,
+    bool* primary_is_top
+) {
+    struct SurfacesValidator {
+        // assuming that samples axis in the file has increasing values
```

Comment:
_positive increasing values_ . Do we assume this anywhere else in the codebase? It's true for 99.999% of files, but there is no guarantee that they are always positive and increasing. We might want a check for this when opening the file